### PR TITLE
Re-add deprecated typography docs

### PR DESCRIPTION
**Pull request by laurkim**
### WHY are these changes introduced?

Re-adds the deprecated typography docs and images on the style guide so that users can still reference them.
The `get-props` script had to be updated to resolve build errors. It was failing since the component files no longer existed in `polaris-react/src/components`.

### WHAT is this pull request doing?

- Re-add dep components images
- Re-add dep component pages
- ~Adds new `deprecatedComponents` array to the `get-props` file and if the parsed file matches any of the elements, return empty object~
- If component status is deprecated, props table is not rendered on component page

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by aveline — APPROVED** [review]

**Review by sam-b-rose — APPROVED** [review]
I like that we provide the mappings to the new components. Should we also include props and usage information until `v12` for those not ready to upgrade quite yet? Looks like GitHub Primer does this by moving the components to a [deprecated folder](https://github.com/primer/react/blob/main/src/deprecated/index.ts#L15) for referencing the documentation, but not exporting from the latest version index.

Maybe this is out of scope for this change. But possibly something to consider for future deprecated components.

File: polaris.shopify.com/scripts/get-props/src/get-props.ts [hunk 1]
```diff
@@ -17,6 +17,15 @@ const compilerOptions = {
...
 };
 
+const deprecatedComponents = [
+  'DisplayText',
+  'Heading',
+  'Subheading',
+  'Caption',
+  'TextStyle',
+  'VisuallyHidden',
+];
```
Comment (alex-page) [inline polaris.shopify.com/scripts/get-props/src/get-props.ts]: This is fine for now but I would prefer us to not hard code an array and get the `deprecated` value from:
```
status:
  value: Deprecated
```

Would this even run if the components don't exist?!

**Review by sarahill — APPROVED** [review]

File: polaris.shopify.com/scripts/get-props/src/get-props.ts (hunk 1 above)
Comment (laurkim) [reply polaris.shopify.com/scripts/get-props/src/get-props.ts]: I originally wanted to status from `frontMatter` but I wasn't sure if us not showing the prop table from all deprecated components and potentially before there's replacement to support functionality would be something we'd want to do. If that's the case, then I'm happy to refactor this 👍 

This does run even if the components don't exist in `polaris-react` but there are examples in `polaris.shopify.com` dir.

**Comment by laurkim** [comment]
> I like that we provide the mappings to the new components. Should we also include props and usage information until `v12` for those not ready to upgrade quite yet? Looks like GitHub Primer does this by moving the components to a [deprecated folder](https://github.com/primer/react/blob/main/src/deprecated/index.ts#L15) for referencing the documentation, but not exporting from the latest version index.
> 
> Maybe this is out of scope for this change. But possibly something to consider for future deprecated components.

I think this is a valid point but may require a bigger refactor of the style guide and increase scope for this change by a significant amount. I'm going to merge this into the v11 branch but happy to continue to have this discussion 👍

File: polaris.shopify.com/scripts/get-props/src/get-props.ts (hunk 1 above)
Comment (alex-page) [reply polaris.shopify.com/scripts/get-props/src/get-props.ts]: Ok that makes sense Lo!

**Pull request merged**